cmake/SetupCodeMetrics.cmake

```diff
@@ -49,7 +49,28 @@
 # Note: Only supported for gnu or clang.
 ########################################
 if (ENABLE_COVERAGE)
-    include(blt/cmake/CodeCoverage.cmake)
+    ##########################################################################
+    # Setup coverage compiler flags 
+    ##########################################################################
+    # Set the actual flags for coverage in the COVERAGE_FLAGS variable 
+    # Note: '--coverage' is equivalent to '-fprofile-arcs -ftest-coverage' for compiling and '-lgcov' for linking
+    # Additional flags that might be useful: 
+    #       " -fno-inline -fno-inline-small-functions -fno-default-inline"
+    blt_append_custom_compiler_flag(FLAGS_VAR   COVERAGE_FLAGS 
+                                    DEFAULT     "--coverage")
```

Comment:
Ideally, it would be good to add the proper cases for the known compilers.

I have only done coverage with gcc and clang, so those are all I can vouch for. 

I am all for adding the correct path for intel in the future. 

Since this is screened by `ENABLE_COVERAGE`,  I think the default is ok until we have more info.